internal/provider/resource_application.go

```diff
@@ -1355,27 +1361,48 @@ func (r *applicationResource) computeExposeDeltas(ctx context.Context, stateExpo
 		return nil, []string{}, diags
 	}
 
-	toExpose := make(map[string]interface{})
-	toUnexpose := make([]string, 0)
+	planEndpoints := strings.Split(planNestedExpose[0].Endpoints.ValueString(), ",")
```

Comment:
Reading this makes me realise the schema for `expose` is slightly wrong, it shouldn't be a list of nested objects, just a single nested object. We only ever seem to use the first item from the list anyway. Maybe we can fix this in here as well before we release 1.0?

Reply:
this will not land in time for 1.0

Reply:
but i do agree the schema for expose is wrong..